packages/@headlessui-react/src/hooks/document-overflow/handle-ios-locking.ts

```diff
@@ -95,6 +95,12 @@ export function handleIOSLocking(): ScrollLockStep<ContainerMetadata> {
             // Check if we are scrolling inside any of the allowed containers, if not let's cancel the event!
             if (e.target instanceof HTMLElement) {
               if (inAllowedContainer(e.target as HTMLElement)) {
+                // Some inputs like `<input type=range>` use touch events to
+                // allow interaction. We should not prevent this event.
+                if (e.target.tagName === 'INPUT') {
+                  return
+                }
```

Comment:
Q: should we hoist this above the `inAllowedContainer` if statement such that we don't have to do that work? Because if it is an `input`, we can bail early and don't need to do any work.

Thoughts?

Reply:
Updated.